staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -135,3 +137,41 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx context.Context, obj runtime.
 	}
 	return true, false, nil
 }
+
+// AdmissionToValidateObjectDeleteFunc returns a admission validate func for object deletion
+func AdmissionToValidateObjectDeleteFunc(admit admission.Interface, staticAttributes admission.Attributes) ValidateObjectFunc {
+	mutatingAdmission, isMutatingAdmission := admit.(admission.MutationInterface)
+	validatingAdmission, isValidatingAdmission := admit.(admission.ValidationInterface)
+
+	return func(old runtime.Object) error {
+		if !isMutatingAdmission && !isValidatingAdmission {
+			return nil
+		}
+		objectMeta, err := meta.Accessor(old)
+		if err != nil {
+			return err
+		}
+		finalAttributes := admission.NewAttributesRecord(
+			nil,
+			old,
+			staticAttributes.GetKind(),
+			staticAttributes.GetNamespace(),
+			objectMeta.GetName(),
```

Comment:
For reviewers, forcing the name via accessor here is because that `DeleteCollections` will set a empty name for the attributes.